diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 4cc86d1..00a2a82 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 2.8.11)
-project(PSRP)
+project(PSRP C)
 
 # Export commands for auto-completion engines
 set(CMAKE_EXPORT_COMPILE_COMMANDS 1)
@@ -13,7 +13,9 @@ add_definitions(-D_GNU_SOURCE)
 find_package(Threads REQUIRED)
 
 # Search OpenSSL
-if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+if (DEFINED ENV{OPENSSL_ROOT_DIR})
+	message(STATUS "Using custom openssl $ENV{OPENSSL_ROOT_DIR}")
+elseif (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
 	set(OPENSSL_ROOT_DIR /usr/local/opt/openssl)
 	find_package(openssl REQUIRED)
 	message(STATUS "Using OpenSSL ${OPENSSL_VERSION}")
